int32[6] measured
int32[6] desired
